### PR TITLE
internal: Build test-macros in a build script

### PR DESCRIPTION
**Pull request by jonas-schievink**
This build the test-proc-macros in `proc_macro_test` in a build script, and copies the artifact to `OUT_DIR`. This should make it available throughout all of rust-analyzer at no cost other than depending on `proc_macro_test`, fixing https://github.com/rust-analyzer/rust-analyzer/issues/9067.

This hopefully will let us later write inline tests that utilize proc macros, which makes my life fixing proc macro bugs easier.

Opening this as a sort of RFC, because I'm not totally sure this approach is the best.

### CODE REVIEW

**Comment by jonas-schievink** [comment]
bors try

**Comment by jonas-schievink** [comment]
bors try

**Comment by jonas-schievink** [comment]
bors try

**Comment by jonas-schievink** [comment]
Okay, looks like d236fc6 fixed the Windows problems I ran into with https://github.com/rust-analyzer/rust-analyzer/pull/9083, nice

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 [workspace]
 resolver = "2"
 members = ["xtask/", "lib/*", "crates/*"]
+exclude = ["crates/proc_macro_test/imp"]
```
Comment (matklad) [inline Cargo.toml:4]: Do we really need this? I think the fact that `imp/Cargo.toml` has `[workspace]` should be enough.

**Review by matklad — COMMENTED** [review]
:+1:

File: Cargo.toml (hunk 1 above)
Comment (jonas-schievink) [reply Cargo.toml:4]: That's what I tried before and it resulted in a Cargo error

`error: multiple workspace roots found in the same workspace`

**Pull request merged**